src/plugins/Tus10.js

```diff
@@ -49,62 +49,21 @@ module.exports = class Tus10 extends Plugin {
 
     this.handlePauseAll = this.handlePauseAll.bind(this)
     this.handleResumeAll = this.handleResumeAll.bind(this)
+    this.handleRetryAll = this.handleRetryAll.bind(this)
     this.handleResetProgress = this.handleResetProgress.bind(this)
     this.handleUpload = this.handleUpload.bind(this)
   }
 
-  pauseResume (action, fileID) {
-    const updatedFiles = Object.assign({}, this.core.getState().files)
-    const inProgressUpdatedFiles = Object.keys(updatedFiles).filter((file) => {
-      return !updatedFiles[file].progress.uploadComplete &&
-             updatedFiles[file].progress.uploadStarted
-    })
-
-    switch (action) {
-      case 'toggle':
-        if (updatedFiles[fileID].uploadComplete) return
-
-        const wasPaused = updatedFiles[fileID].isPaused || false
-        const isPaused = !wasPaused
-        let updatedFile
-        if (wasPaused) {
-          updatedFile = Object.assign({}, updatedFiles[fileID], {
-            isPaused: false
-          })
-        } else {
-          updatedFile = Object.assign({}, updatedFiles[fileID], {
-            isPaused: true
-          })
-        }
-        updatedFiles[fileID] = updatedFile
-        this.core.setState({files: updatedFiles})
-        return isPaused
-      case 'pauseAll':
-        inProgressUpdatedFiles.forEach((file) => {
-          const updatedFile = Object.assign({}, updatedFiles[file], {
-            isPaused: true
-          })
-          updatedFiles[file] = updatedFile
-        })
-        this.core.setState({files: updatedFiles})
-        return
-      case 'resumeAll':
-        inProgressUpdatedFiles.forEach((file) => {
-          const updatedFile = Object.assign({}, updatedFiles[file], {
-            isPaused: false
-          })
-          updatedFiles[file] = updatedFile
-        })
-        this.core.setState({files: updatedFiles})
-    }
-  }
-
   handlePauseAll () {
-    this.pauseResume('pauseAll')
+    this.core.pauseResume('pauseAll')
   }
 
   handleResumeAll () {
-    this.pauseResume('resumeAll')
+    this.core.pauseResume('resumeAll')
+  }
+
+  handleRetryAll () {
+    this.core.pauseResume('retryAll')
   }
```

Comment:
I guess these pause-all/resume-all/retry-all event handlers could also be moved to core

Reply:
Maybe it would make sense to have pauseAll, resumeAll, retryAll methods in Core, that do what the pauseResume method does now, and then fire their event? Then pressing the retry button would do `this.core.retryAll()` instead of `this.core.emit('core:retry-all')`.

Reply:
Refactored it! Please beware that this has changed :)